src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -579,14 +600,24 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 			// Override handling: pick editor or open specific
 			if (resolvedOptions?.override === EditorOverride.PICK || typeof resolvedOptions?.override === 'string') {
-				return this.openEditorWith(resolvedOptions.override, resolvedEditor, resolvedOptions, resolvedGroup);
+				const resolvedInputWithOptions = await this.editorOverrideService.resolveEditorOverride(editor as IEditorInput, resolvedOptions, resolvedGroup);
+				if (!resolvedInputWithOptions) {
+					return;
+				}
+				return resolvedGroup.openEditor(resolvedInputWithOptions.editor, resolvedInputWithOptions.options ?? resolvedOptions);
 			}
 
 			// Override handling: ask providers to override
 			if (resolvedOptions?.override !== EditorOverride.DISABLED) {
-				const override = this.doOverrideOpenEditor(resolvedEditor, resolvedOptions, resolvedGroup);
-				if (override) {
-					return override;
+				// This will get cleaned up soon, but since the override service no longer uses the override flow we must check that
```

Comment:
There seems to be some overlap with the above code, but you seem to indicate that this is being cleaned up, maybe add a `TODO` to not forget.

Reply:
Yeah I could clean up this patch a bit, but most of this will go away once everyone adopts it.